visualization/dashboard/src/MLIDashboard/Controls/CohortEditor/CohortEditor.tsx

```diff
@@ -171,16 +182,28 @@ export class CohortEditor extends React.PureComponent<ICohortEditorProps, ICohor
                             }
                         </div>
                     </div>
-                    <PrimaryButton onClick={this.saveCohort} className={styles.saveCohort}>Save</PrimaryButton>
+                    {this.props.isNewCohort ?
+                        <PrimaryButton onClick={this.saveCohort} className={styles.saveCohort}>{localization.CohortEditor.save}</PrimaryButton>
+                        :
+                        <div className={styles.saveAndDeleteDiv}>
+                            <DefaultButton onClick={this.deleteCohort.bind(this)} className={styles.deleteCohort}>{localization.CohortEditor.delete}</DefaultButton>
+                            <PrimaryButton onClick={this.saveCohort} className={styles.saveCohort}>{localization.CohortEditor.save}</PrimaryButton>
+                        </div>
+                    }
                 </div>
             </Callout>
         );
     }
 
     private closeCallout = (): void => {
+        this.setState({ openedFilter: undefined, filters: undefined, filterIndex: undefined})
```

Comment:
interesting, were you seeing state being persisted across uses of the callout? there is a parameter on Fabric callouts for if they should be persisted vs created new each time, that would probably be the way to solve this, since doing it this way will require clearing the state on delete and on save, or any other way of exiting.

Reply:
Hmm, I assumed it was good practice to reset states before closing the callout so I never ran into persisted state issue. I'll take a look at the parameter to see if its needed here.